lib/v4/migration-helpers/update-api-folder-structure/index.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * Migrate API folder structure to v4
+ */
+
+const { resolve, join, basename } = require('path');
+const fs = require('fs-extra');
+const _ = require('lodash');
+const chalk = require('chalk');
+
+const pluralize = require('pluralize');
+const runJscodeshift = require('../../utils/run-jscodeshift');
+const { logger } = require('../../../global/utils');
+const updateContentTypes = require('../convert-models-to-content-types');
+const updateRoutes = require('../update-routes');
+const updatePolicies = require('../update-api-policies');
+const renameApiFilesToSingular = require('../rename-api-files-to-singular');
+const { getDirsAtPath, cleanEmptyDirectories } = require('./utils');
+
+const updateApiFolderStructure = async (appPath) => {
+  const strapiAppPath = resolve(appPath);
+  const apiDirCopyPath = join(strapiAppPath, 'src', 'api');
+
+  try {
+    // Copy the folder to a v3 folder for safe keeping
+    await fs.copy(join(strapiAppPath, 'api'), join(strapiAppPath, 'v3', 'api'));
+    // Move the original api folder to src/api
+    await fs.move(join(strapiAppPath, 'api'), apiDirCopyPath);
```

Comment:
ℹ️ Hard to see the diff because I renamed the file. Make a copy of the api folder, and store it along with all other files in a v3 directory. Then move the original to the src dir for v4